### ISSUE TITLE
Bug - Build fails when using Version builder plugin and project contains method extending enum

### ISSUE THREAD

**Issue by Legoldos** [EWSoftware/SHFB#1071]
Hello I have encountered issue in sandcastle version 2024.2.18.0. 

My motivation:
I have multiple projects targeting different frameworks and I want to generate single help file for all of them, thats why I am using Version builder plug-in

Setup:
Create two projects in C# - class library .net 8.0 (framework version doesnt seem to matter)
first project is empty
second project contains this class
```
using Microsoft.Extensions.Logging;

namespace ProjectWithEnum
{
    public enum LogMessageID
    {
        Normal = 0
    }


    public static class LoggingExtensions
    {

        /// <summary>
        /// This throws error when build
        /// </summary>
        public static void ExtendingProjectDefinedEnum(this LogMessageID logMessageID)
        {
        }

        /// <summary>
        /// This builds ok
        /// </summary>
        public static void ExtendingEnumDefinedOutside(this LogLevel logLevel)
        {
        }
    }
}

```

Create 2 fresh .shfbproj projects 
Add documentation source to each project (one will contain empty csproj second will contain csproj with enum)
Add version builder plugin to one of the project (it doesnt matter which one) and add other project 

Build shfbproj project containing version builder

following error occurs

```
SHFB: Error BE0065: BUILD FAILED: TransformComponent: [T:Projectlatest.LogMessageID] An error occurred while attempting to transform the reflection data to a topic. The error message was: System.NullReferenceException: Odkaz na objekt není nastaven na instanci objektu.
   v Sandcastle.PresentationStyles.Default2022.Default2022Transformation.<>c__DisplayClass75_0.<RenderApiEnumerationMembersList>b__0(XElement el)
   v System.Linq.EnumerableSorter`2.ComputeKeys(TElement[] elements, Int32 count)
   v System.Linq.EnumerableSorter`1.Sort(TElement[] elements, Int32 count)
   v System.Linq.OrderedEnumerable`1.<GetEnumerator>d__1.MoveNext()
   v System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   v System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   v Sandcastle.PresentationStyles.Default2022.Default2022Transformation.RenderApiEnumerationMembersList(TopicTransformationCore transformation)
   v Sandcastle.PresentationStyles.Default2022.Default2022Transformation.RenderTopic()
   v Sandcastle.Tools.BuildComponents.TransformComponent.Apply(XmlDocument document, String key)
   v SandcastleBuilder.Utils.BuildEngine.BuildAssemblerInternal.WriteMessage(String componentName, MessageLevel level, String key, String message)
   v Sandcastle.Core.BuildAssembler.BuildComponent.BuildComponentCore.WriteMessage(String key, MessageLevel level, String message, Object[] args)
   v Sandcastle.Tools.BuildComponents.TransformComponent.Apply(XmlDocument document, String key)
   v Sandcastle.Tools.BuildComponents.SwitchComponent.Apply(XmlDocument document, String key)
   v SandcastleBuilder.Utils.BuildEngine.BuildAssemblerInternal.Apply(IEnumerable`1 topics)
   v SandcastleBuilder.Utils.BuildEngine.BuildAssemblerInternal.<BuildTopics>b__27_0()
   v System.Threading.Tasks.Task.Execute()

Došlo k jedné nebo více chybám.
   v System.Threading.Tasks.Task.WaitAll(Task[] tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken)
   v SandcastleBuilder.Utils.BuildEngine.BuildAssemblerInternal.BuildTopics()
   v SandcastleBuilder.Utils.BuildEngine.BuildProcess.Build()

TransformComponent: [T:Projectlatest.LogMessageID] An error occurred while attempting to transform the reflection data to a topic. The error message was: System.NullReferenceException: Odkaz na objekt není nastaven na instanci objektu.
   v Sandcastle.PresentationStyles.Default2022.Default2022Transformation.<>c__DisplayClass75_0.<RenderApiEnumerationMembersList>b__0(XElement el)
   v System.Linq.EnumerableSorter`2.ComputeKeys(TElement[] elements, Int32 count)
   v System.Linq.EnumerableSorter`1.Sort(TElement[] elements, Int32 count)
   v System.Linq.OrderedEnumerable`1.<GetEnumerator>d__1.MoveNext()
   v System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   v System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   v Sandcastle.PresentationStyles.Default2022.Default2022Transformation.RenderApiEnumerationMembersList(TopicTransformationCore transformation)
   v Sandcastle.PresentationStyles.Default2022.Default2022Transformation.RenderTopic()
   v Sandcastle.Tools.BuildComponents.TransformComponent.Apply(XmlDocument document, String key)
   v SandcastleBuilder.Utils.BuildEngine.BuildAssemblerInternal.WriteMessage(String componentName, MessageLevel level, String key, String message)
   v Sandcastle.Core.BuildAssembler.BuildComponent.BuildComponentCore.WriteMessage(String key, MessageLevel level, String message, Object[] args)
   v Sandcastle.Tools.BuildComponents.TransformComponent.Apply(XmlDocument document, String key)
   v Sandcastle.Tools.BuildComponents.SwitchComponent.Apply(XmlDocument document, String key)
   v SandcastleBuilder.Utils.BuildEngine.BuildAssemblerInternal.Apply(IEnumerable`1 topics)
   v SandcastleBuilder.Utils.BuildEngine.BuildAssemblerInternal.<BuildTopics>b__27_0()
   v System.Threading.Tasks.Task.Execute()
```

If I disable version builder pluing or build just the project with enum then build is completed without error.

Also if I comment method ExtendingProjectDefinedEnum and try to build documentation just for method ExtendingEnumDefinedOutside then the build is also OK.

Any idea if the issue is in my configuration or its in sandcastle?

**Comment by BlackyTheDruide** [comment]
Hi,

It seems that this is a common problem in Sandcastle, because we are facing exactly the same problem. Like @Legoldos, we have a setup with many projects that need to be part of one documentation.

We have not been able to find a workaround for this issue, so in our opinion this needs to be fixed on Sandcastle's end.

Best regards

**Comment by EWSoftware** [comment]
It's caused by the enumeration member list page not handling extension methods correctly.  I have a fix and will probably be able to get a new release out in a week or so.  In the meantime, you can work around it by excluding the extension methods that extend an enumerated type.

**Comment by BlackyTheDruide** [comment]
@EWSoftware Thank you for the information. Unfortunately we have to many projects to fix this on our code side, so we need to wait until a new version including the fix is available.

**Comment by EWSoftware** [comment]
The new release is available now (v2024.9.12.0).
